File: bindings/nodejs/types/secretManager.ts
/** Secret manager that uses a Ledger Nano hardware wallet or Speculos simulator. */
export interface LedgerNanoSecretManager {
    /** boolean indicates whether it's a simulator or not. */
    LedgerNano: boolean;
}

/** Secret manager that uses a mnemonic. */
export interface MnemonicSecretManager {
    Mnemonic: string;
}

/** Secret manager that uses a seed. */
export interface SeedSecretManager {
    HexSeed: string;
}

/** Secret manager that uses Stronghold. */
export interface StrongholdSecretManager {
    Stronghold: {
        password?: string;
        snapshotPath?: string;
    };
}

/** The status of a Ledger Nano */
export interface LedgerNanoStatus {
    connected: boolean;
    locked: boolean;
    blindSigningEnabled: boolean;
    app?: LedgerApp;
    device?: LedgerDeviceType;
    bufferSize?: number;
}

/** The current opened app */
export interface LedgerApp {
    name: string;
    version: string;
}

/** The Ledger Device Type */
export enum LedgerDeviceType {
    LedgerNanoS = 'LedgerNanoS',
    LedgerNanoX = 'LedgerNanoX',
    LedgerNanoSPlus = 'LedgerNanoSPlus',
}

/** Supported secret managers */
export type SecretManager =
    | LedgerNanoSecretManager
    | MnemonicSecretManager
    | StrongholdSecretManager;

File: bindings/python/native/iota_wallet/secret_manager.py
class LedgerNanoSecretManager(dict):
    def __init__(self, is_simulator):
        """Initialize a ledger nano secret manager.
        """

        dict.__init__(self, ledgerNano=is_simulator)


class MnemonicSecretManager(dict):
    def __init__(self, mnemonic):
        """Initialize a mnemonic secret manager.
        """

        dict.__init__(self, mnemonic=mnemonic)


class SeedSecretManager(dict):
    def __init__(self, seed):
        """Initialize a seed secret manager.
        """

        dict.__init__(self, hexSeed=seed)


class StrongholdSecretManager(dict):
    def __init__(self, snapshot_path, password):
        """Initialize a stronghold secret manager.
        """

        dict.__init__(self, stronghold=StrongholdSecretManager.Inner(
            snapshot_path, password))

    class Inner(dict):
        def __init__(self, snapshot_path, password):
            dict.__init__(self, password=password, snapshotPath=snapshot_path)
